File: 2859/go/main.go
package main

func sumIndicesWithKSetBits(nums []int, k int) int {
	result, l := 0, len(nums)
	for i := 0; i < l; i++ {
		oneCount, num := 0, i
		for num > 0 {
			num &= num - 1
			oneCount++
		}
		if oneCount == k {
			result += nums[i]
		}
	}
	return result
}

func main() {
	println(sumIndicesWithKSetBits([]int{5, 10, 1, 5, 2}, 1))
	println(sumIndicesWithKSetBits([]int{4, 3, 2, 1}, 2))
}

File: 2859/python3/main.py
class Solution:
    def sumIndicesWithKSetBits(self, nums: list[int], k: int) -> int:
        result = 0
        for i, num in enumerate(nums):
            if bin(i).count("1") == k:
                result += num
        return result


s = Solution()

print(s.sumIndicesWithKSetBits([5, 10, 1, 5, 2], 1))
print(s.sumIndicesWithKSetBits([4, 3, 2, 1], 2))
